Frontend: Upgrade Vuetify from v1.5 to v3.7

AsikNasik opened the pull request:
<!--

Thank you for your interest in contributing!

Because we want to create the best possible product for our users, we have a set of criteria to ensure that all submissions are acceptable, see https://docs.photoprism.app/developer-guide/pull-requests/ for details.

(1) Please provide a concise description of your pull request.

- What does it implement / fix / improve? Why?
- Are the changes related to an existing issue?

(2) After you submit your first pull request, you will be asked to accept our CLA, see https://www.photoprism.app/cla.

(3) Finally, please confirm that the following criteria are met by replacing "[ ]" with "[x]" (also possible at a later time).

-->

Acceptance Criteria:

- [ ] Features and enhancements must be fully implemented so that they can be released at any time without additional work
- [ ] Automated unit and/or acceptance tests are mandatory to ensure the changes work as expected and to reduce repetitive manual work
- [ ] Frontend components must be responsive to work and look properly on phones, tablets, and desktop computers; you must have tested them on all major browsers and different devices
- [ ] Documentation and translation updates should be provided if needed
- [ ] In case you submit database-related changes, they must be tested and compatible with SQLite 3 and MariaDB 10.5.12+

<!--

Since reviewing, testing and finally merging pull requests requires significant resources on our side, this can take several months if it's not just a small fix, especially if extensive testing is required to prevent bugs from getting into our stable version.

We thank you for your patience! :)

-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/photoprism/photoprism?pullRequest=4556) <br/>All committers have signed the CLA.

lastzero commented on frontend/package.json:
```diff
@@ -119,5 +119,9 @@
+  ],
+  "devDependencies": {
+    "@mdi/font": "^7.4.47",
+    "eslint-plugin-vuetify": "^2.4.0"
```
You're welcome to move the other eslint packages to devDependencies as well, if that has an advantage?

So far my understanding is that it makes no difference for our use case, as **npm** is only used for development, while the production build only contains precompiled JS/CSS files and therefore never contains any dev dependencies (whether they are listed as such or not)...

AsikNasik replied:
Okay, I will check it later if it has any sense to do it. For now I propose to focus to the migration and after I will do some investigation about this question. What do you think about it?

lastzero commented on frontend/package.json:
```diff
@@ -119,5 +119,9 @@
-  ]
+  ],
+  "devDependencies": {
+    "@mdi/font": "^7.4.47",
```
Due to some breaking changes in the [material icons font](https://www.npmjs.com/package/material-design-icons-iconfont) that we use, we added the required CSS and font files to our own repository under [`/frontend/src/css/vendor/icons`](https://github.com/photoprism/photoprism/tree/develop/frontend/src/css/vendor/icons) (AFAIK, the original package no longer included a plain CSS file or the font paths in it were invalid, so the icons broke in the UI after updating the package):
- https://www.npmjs.com/package/material-design-icons-iconfont
- https://jossef.github.io/material-design-icons-iconfont/

Instead of using/importing the original npm package, `/frontend/src/css/vendor/iconsmaterial-design-icons.css` is imported in our main CSS file:
https://github.com/photoprism/photoprism/blob/0b391d26ecf3989becef2b8280a15b0014e709f2/frontend/src/css/app.css#L26

If Vuetify 2/3 use `@mdi/font` instead of `material-design-icons-iconfont` then that's fine with us as both should support/include similar icons since they are also based on Material Design:
- https://fonts.google.com/icons

Note, however, that the naming of some icons MAY be different and that you will therefore probably need to change some templates so they use the new Material Design font icons:
- https://www.npmjs.com/package/@mdi/font
- https://pictogrammers.com/library/mdi/

Also, would it make sense to list `@mdi/font` under the regular `dependencies` in the `package.json` file, since the fonts are always required? See the comment below.

AsikNasik replied:
Yes, you are right.
I will move `@mdi/font` package to the `dependencies` because we always need them.

lastzero reviewed (commented):
Here is some initial feedback for you, mainly regarding the icons. Since these are important e.g. for the navigation, it might make sense to wait with further tests until your work on them is finished?

AsikNasik commented on frontend/src/css/themes.css:
Will we keep this comment or will delete in the future?

AsikNasik commented on frontend/src/css/themes/default.css:
```diff
@@ -1,5 +1,5 @@
 /* Component Color Defaults */
-
+/*
```
Will we keep this comment or will delete in the future?

lastzero committed:
FFmpeg: Refactor internal/ffmpeg/transcode_cmd.go #4604

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
About: Update copyright notices

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Server: Add link to documentation for unix domain sockets #4673 #4684

see https://github.com/photoprism/photoprism/discussions/4710

Signed-off-by: Michael Mayer <michael@photoprism.app>

graciousgrey committed:
Frontend: Fix icon #3168

lastzero committed:
Viewer: Add Type, Lat, Lng, and Duration to model data #1307 #4698 #4708

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update deps in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Add media codec to API response data #1307 #4698 #4708

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Fix typo in photo search result structs #1307 #4698 #4708

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Update Go dependencies in go.mod and go.sum

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update deps in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

AsikNasik committed:
Viewer: Add toggle to flag/unflag pictures as favorite #4700 #4709

lastzero committed:
Viewer: Add Toggle to Select/Unselect Pictures #4699 #4700 #4709

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Add pswp__icn-shadow to improve icon contrast #4699 #4700 #4709

Signed-off-by: Michael Mayer <michael@photoprism.app>

omerdduran committed:
Viewer: Improve loading of videos, animations, and live photos #1307 #3372 #4698 #4702 #4708

* Frontend: Refactor video element creation #4698

* Frontend: Enable autoplay for Live Photos and GIFs with looping functionality #4698

* Viewer: Media type checks  #4698

* Viewer: Removed unnecessary API call

lastzero committed:
Viewer: Refactor loading of playable content #1307 #4698 #4708

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Update dynamic caption styles in viewer.css #1307

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Fix toggle icon shadow in Apple Safari #4699 #4700 #4709

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Fix double </use> tag in component/viewer.vue #4699 #4700 #4709

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Add fullscreen mode toggle control #1307 #3168 #4703

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update deps in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Add slideshow toggle control #1307 #3168 #4702 #4703

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Viewer: Improve padding based on image size and screen resolution #4704

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Remove "v-col-xxl-1" break point in view/cards.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

# Conflicts:
#	frontend/src/locales/hi.po

lastzero committed:
Frontend: Update translations.json in /src/locales #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Remove .pswp__dynamic-caption transition #1307 #3168 #4702 #4703

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Reduce spacing in delete and remove dialogs #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Append "content-copy" icons in dialog/webdav.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Refactor service edit dialog in service/dialog/edit.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Allow file/photo metadata to be wrapped in photo edit dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Upgrade "github.com/gin-contrib/gzip" in go.mod and go.sum

Signed-off-by: Michael Mayer <michael@photoprism.app>

graciousgrey committed:
Frontend: Improve theme colors #3168

AsikNasik committed:
Viewer: add generating srcset and checking data on imageSizeChange #4704

AsikNasik committed:
Viewer: fix console log with image source when change size #4704

omerdduran committed:
fix: remove duplicate hash in Yellowstone theme button color

omerdduran committed:
fix: remove duplicate hash in Yellowstone theme button color

omerdduran committed:
Viewer: Implement dynamic thumbnail loading on zoom level change

graciousgrey committed:
Frontend: Improve theme colors #3168

graciousgrey committed:
Frontend: Improve theme colors #3168

lastzero committed:
UX: Add "fill-opacity" theme variable #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Add "fill-opacity" theme variable #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

weblate committed:
Weblate: Update frontend translations

omerdduran committed:
fix: Resolve unresolved pswp variable

AsikNasik committed:
Viewer: add checking of actual width/height and  next thumbnail #4704

omerdduran committed:
fix: resolve linting error

AsikNasik committed:
Viewer: delete unused console.logs #4704

omerdduran committed:
Viewer: Refactor slide content update

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Backend: Upgrade "github.com/abema/go-mp4" in go.mod and go.sum

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Upgrade "github.com/abema/go-mp4" in go.mod and go.sum

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Refactor thumb URL generation in the backend API #4704 #4722

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Refactor thumb URL generation in the backend API #4704 #4722

Signed-off-by: Michael Mayer <michael@photoprism.app>

omerdduran committed:
Merge remote-tracking branch 'origin/upgrade/vuetify-3' into feature/Dynamically-set-image-src,-width-and-height-to-match-current-zoom-level-and-viewport-#4704

lastzero committed:
UX: Add theme color variable for switch input elements #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reformat options/themes.js #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Show all rows in the app passwords dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

omerdduran committed:
Viewer: Optimizing URL checks for loading higher quality images

lastzero committed:
Viewer: Only open as stack if there are multiple JPEG files #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Refactor handleZoomLevelChange() function #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Refactor imports in src/app.js #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

graciousgrey committed:
Frontend: Improve theme colors #3168

lastzero committed:
Frontend: Refactor import of src/component/about/footer.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Change import order in src/app.js #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update dependencies in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

omerdduran committed:
Viewer: simplifying URL size comparison logic

lastzero committed:
Frontend: Refactor src/app.js and components.js imports #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Change import order in component/components.js imports #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Refactor handleZoomLevelChange() function #3168 #4704

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Update "github.com/davidbyttow/govips/v2" in go.mod and go.sum

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Fix category input in src/album/dialog/edit.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Replace <translate> tags in page/about/about.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Replace deprecated <translate> tags with $gettext() #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

# Conflicts:
#	frontend/src/locales/af.po
#	frontend/src/locales/ar.po
#	frontend/src/locales/be.po
#	frontend/src/locales/bg.po
#	frontend/src/locales/ca.po
#	frontend/src/locales/cs.po
#	frontend/src/locales/da.po
#	frontend/src/locales/de.po
#	frontend/src/locales/el.po
#	frontend/src/locales/en.po
#	frontend/src/locales/es.po
#	frontend/src/locales/et.po
#	frontend/src/locales/eu.po
#	frontend/src/locales/fa.po
#	frontend/src/locales/fi.po
#	frontend/src/locales/fr.po
#	frontend/src/locales/ga.po
#	frontend/src/locales/he.po
#	frontend/src/locales/hr.po
#	frontend/src/locales/hu.po
#	frontend/src/locales/id.po
#	frontend/src/locales/it.po
#	frontend/src/locales/ja.po
#	frontend/src/locales/ko.po
#	frontend/src/locales/ku.po
#	frontend/src/locales/lt.po
#	frontend/src/locales/ms.po
#	frontend/src/locales/nb.po
#	frontend/src/locales/nl.po
#	frontend/src/locales/pl.po
#	frontend/src/locales/pt.po
#	frontend/src/locales/pt_BR.po
#	frontend/src/locales/ro.po
#	frontend/src/locales/ru.po
#	frontend/src/locales/sk.po
#	frontend/src/locales/sl.po
#	frontend/src/locales/sv.po
#	frontend/src/locales/th.po
#	frontend/src/locales/tr.po
#	frontend/src/locales/translations.pot
#	frontend/src/locales/uk.po
#	frontend/src/locales/vi.po
#	frontend/src/locales/zh.po
#	frontend/src/locales/zh_TW.po

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update dependencies in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reduce eslint prettier printWidth to 120 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reformat component/navigation.vue and viewer.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Router: Fix requiresAuth check in frontend/src/app.js #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Commands: Remove unused faceFlagsInfo var from show_config_options.go

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Commands: Fix config options report in config/cli_flags_report.go

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Improve readability of text links e.g. on about page #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Remove obsolete theme stylesheets in /src/css/themes #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Replace "number" output filter with $util.formatNs() #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
API: Add additional fields to label and subject edit forms #383 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Labels: Refactor label entity and cache in label.go and label_cache.go

Signed-off-by: Michael Mayer <michael@photoprism.app>

omerdduran committed:
fix: enable pointer events for caption text while keeping overlay non-interactive

lastzero committed:
Metadata: Set labels based on matching keywords in the caption #4603

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Metadata: Increase probability of keywords extracted from captions #4603

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Don't loop videos when a slideshow is playing #1307 #4698 #4702

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Settings: Refactor services.vue and service add/edit dialogs #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
People: Fix edit dialog save button in people/dialog/edit.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Develop: Upgrade base image from 250103-oracular to 250118-oracular

Includes the latest stable version of Go.

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Entity: Add json annotations to Service struct in service.go

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Upgrade Go deps in go.mod and go.sum

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
API: Add Swagger annotations to /api/v1/services endpoints

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Places: Show error message if browser does not support WebGL #4733

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Places: Add documentation link to frontend/src/page/places.vue #4733

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Sharing: Fix copy to clipboard functionality in dialog/share.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
API: Add "@Accept json" Swagger doc annotations where applicable

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Enlarge edit dialog tabs and display badges on small screens #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve label table and input styles in photo edit dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Translate label sources in photo edit dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update src/locales/translations.json

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update dependencies in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Use v-expand-transition for expanded search toolbar #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reformat .vue templates in /src/component/people #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reformat .vue templates in /src/component/photo #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reformat .vue templates in /src/component/label #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reformat .vue templates in /src/component/file #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reformat .vue templates in /src/component/account #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Reformat .vue templates in /src/component/album #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Automatically update people tab in photo edit dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Update dialog styles in src/css/vuetify.css

lastzero committed:
Sharing: Fix "states" link in sidebar navigation #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Upgrade deps in package.json and package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Metadata: Add backward compatibility for Description field #4603

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Places: Add GlobeControl #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Places: Enable 3D GlobeControl for all maps #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Places: Update map background color in css/places.css #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Places: Update map background color in css/places.css #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Scans: Detect Reflecta DigitDia 7000 as film scanner #4581

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Forms UX: Change validate-on="blur" to validate-on="invalid-input" #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Display photo edit dialog as side panel on large screens #3168

This will also open the Files tab when users click on a filename in the
cards view.

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update deps in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Show Title and Caption inputs first in photo/edit/details.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Don't show upload dialog in full-screen mode on large screens #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Change label and icon of upload button in dialog/upload.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Increase minimum column size in photo/edit/people.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Set lg column width in photo/edit/people.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Fix styles and simplify grid view in photo/edit/people.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Ensure new face count is displayed on load in page/people.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Rename "States" to "Regions" in component/navigation.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve table button and photo edit dialog styles #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Remove elevation from photo edit dialog tabs #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve .v-dialog--sidepanel styles in src/css/vuetify.css #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Reformat and improve footer component and about page #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Replace inline style attributes with ".not-selectable" #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Replace <v-container> with plain <div> elements #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

graciousgrey committed:
Annotations: Improve swagger annotations

lastzero committed:
Frontend: Update dependencies in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

graciousgrey committed:
Frontend: Add default value for feedback form

lastzero committed:
Frontend: Ensure that ESLint is installed and works as intended #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Assets: Add mobile-web-app-capable meta tag to HTML templates #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Refactor form fields in page/settings/account.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Refactor tabs and reformat .vue templates with eslint #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update dependencies in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Set htmlWhitespaceSensitivity to "css" in .eslintrc.js #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve usability and mobile friendliness of photo edit dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Stretch Title and Caption input fields in photo edit dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

AsikNasik committed:
Frontend: delete prop required #3168

lastzero committed:
UX: Improve usability of index page, upload page and upload dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update dependencies in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update assertions in tests/unit/options/options_test.js #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Increase <v-badge> padding in vuetify.css #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Fix states navigation link in share view #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update dependencies in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Upgrade "github.com/go-co-op/gocron" in go.mod and go.sum #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Change test URL in internal/form/service_test.go

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Develop: Update NodeJS deps in /scripts/dist/install-nodejs.sh

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Change icon on import button in page/library/import.vue

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve layout of upload dialog and library pages #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve layout of photo edit dialog tabs #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Develop: Upgrade base image from 250118 to 250124-oracular #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Shorten info text in dialog/upload.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update /src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Fix album chip styles in dialog/upload.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Remove "Done" action button from photo/edit/details.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Shorten input placeholder in upload.vue and album/dialog.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update /src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Use icons for country, camera & lens in photo/edit/details.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Change icon in photo/album/dialog.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Automatically expand first link in dialog/share.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Set matching type for video source element #1307 #3168 #4704

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Check support before adding "nodownload" crtl #1307 #3168 #4704

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Live Photos: Set matching type for video source element #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Use click to close edit dialog if the model data is unchanged #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Refactor component names and directory structure #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Update src/component/defaults.js #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Upgrade deps in package.json and package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Shorten text in page/library/index.vue and import.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update /src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Disable album combobox while data is not yet loaded #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve adding names to people in photo/edit/dialog.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Refactor callback names in photo/edit/dialog.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve setting and updating names in page/people/new.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update /src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Set native video stream src based on mimetype #1307 #3168 #4698

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Activate Approve button in edit dialog if photo is in review #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Prevent duplicate albums from being created #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Rename @cancel events to @close in dialog components #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Metadata: Update special words list in /pkg/text/specialwords.go

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Backend: Update Go deps in go.mod and go.sum

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Optimize styles for video playback on mobile devices #4698 #4738

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Resize lightbox after toggling fullscreen mode #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Disable fullscreen mode toggle on mobile devices #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve mobile device detection #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

graciousgrey committed:
Tests: Adapt acceptance tests to changes #3168

graciousgrey committed:
Tests: Adapt acceptance tests to changes #3168

graciousgrey committed:
Tests: Adapt acceptance tests to changes #3168

lastzero committed:
Viewer: Improve playback of videos on mobile devices #3168 #4698

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Toggle controls if not a video and space is pressed #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve touch navigation behavior on mobile devices #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Refactor video formats and codecs in front and backend #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve tooltips in the photo edit dialog #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Metadata: Add "Etc/GMT" time zone test cases to entity_time_test.go

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve mobile layout of Details and Labels in the edit dialog #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
ACL: Update checks and presets #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
UX: Improve form validation in page/settings/account.vue #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update /src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Forms: Move field validation rules to /frontend/src/common/form.js #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Gettext: Replace %{name} with %{s} to avoid translation errors #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update /src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Frontend: Improve form validation in user add and edit dialogs #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Frontend: Update /src/locales/translations.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
ACL: Update test assertions in /internal/auth/acl/acl_test.go

Signed-off-by: Michael Mayer <michael@photoprism.app>

AsikNasik committed:
Frontend: delete not necessary TODOs #3168

AsikNasik committed:
Frontend: change mask property #3168

lastzero committed:
Videos: Enhance support for direct streaming of HEVC / H.265 #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Videos: Improve MP4 AVC browser playback compatibility #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Adjust UI styles for right-to-left languages #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Viewer: Reduce preloading to one slide in each direction #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

graciousgrey committed:
Frontend: Update language string in options.js

lastzero committed:
Videos: Refactor VP8 and VP9 codec types and add codec aliases #1307

Signed-off-by: Michael Mayer <michael@photoprism.app>

graciousgrey committed:
Tests: Adapt acceptance tests to changes #3168

AsikNasik committed:
Frontend: delete TODO #3168

lastzero committed:
Frontend: Update deps in package-lock.json #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
API: Fix content type header for transcoded videos #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
API: Improve content type string generation #1307 #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

lastzero committed:
Merge branch 'develop' into upgrade/vuetify-3

lastzero committed:
Metadata: Improve media type detection in pkg/fs/mime.go #3168

Signed-off-by: Michael Mayer <michael@photoprism.app>

dannyadair commented:
What an awesome effort!
